.github/workflows/deploy.yaml

```diff
@@ -29,45 +31,144 @@ jobs:
         with:
           path: ${{ github.event.repository.name }}
 
-      - name: Overwrite the global Rprofile ✍️
+      - name: Install jq
+        run: sudo apt-get update && sudo apt-get install -y jq
+
+      - name: Extract R version from lockfile
+        run: printf 'R_VERSION=%s\n' "$(jq --raw-output .R.Version ${{ github.event.repository.name }}/${{ matrix.directory }}/renv.lock)" >> $GITHUB_ENV
+
+      - name: Check if cypress test exists
+        id: find-cypress
+        run: |
+          if [ -d ${{ github.event.repository.name }}/${{ matrix.directory }}/tests/cypress ]; then
+            echo "has-cypress-tests=true" >> $GITHUB_OUTPUT
+          else
+            echo "Skipping frontend tests because cypress directory does not exist."
+            echo "has-cypress-tests=false" >> $GITHUB_OUTPUT
+          fi
+
+      - name: Setup system dependencies for cypress and python app
+        if: steps.find-cypress.outputs.has-cypress-tests == 'true'
+        run: >
+          apt-get update && apt-get install --yes
+          libgtk2.0-0 libgbm-dev libnotify-dev libgconf-2-4 xvfb python3.10-venv
+
+      - name: Setup Node
+        uses: actions/setup-node@v3
+        if: steps.find-cypress.outputs.has-cypress-tests == 'true'
+        with:
+          node-version: 16
+
+      - name: Setup npm packages
+        if: steps.find-cypress.outputs.has-cypress-tests == 'true'
+        shell: bash
+        working-directory: ${{ github.event.repository.name }}/${{ matrix.directory }}/js
+        run: |
+          npm i
+
+      - name: Restore renv from cache
+        uses: actions/cache@v2
+        env:
+          CACHE_KEY: renv-cache-${{ runner.arch }}-${{ runner.os }}-${{ env.R_VERSION }}-${{ matrix.directory }}-${{ env.CHANNEL }}
+        with:
+          path: ${{ github.event.repository.name }}/${{ matrix.directory }}/renv/library
+          key: ${{ env.CACHE_KEY }}-${{ hashFiles(format('{0}/{1}/renv.lock', github.event.repository.name, matrix.directory)) }}
+          restore-keys: ${{ env.CACHE_KEY }}-
+
+      - name: Update renv lock file with repo channel
+        shell: Rscript {0}
+        working-directory: ${{ github.event.repository.name }}
+        run: |
+          setwd("${{ matrix.directory }}")
+          repos <- renv::lockfile_read()$R$Repositories
+          repos$Pharmaverse <- "${{ matrix.repository }}"
+          renv::lockfile_modify(repos = repos) |> renv::lockfile_write()
+
+      - name: Install R packages using renv and update the renv snapshot
+        shell: Rscript {0}
+        working-directory: ${{ github.event.repository.name }}/${{ matrix.directory }}
         run: |
-          cat > /usr/local/lib/R/etc/Rprofile.site <<RPROFILE
-           options(
+          options(renv.config.cache.symlinks = FALSE)
+          lockfile_packages <- renv::lockfile_read()$Packages |> names()
+          renv::update(lockfile_packages)
+          renv::snapshot(packages = lockfile_packages, update = TRUE)
```

Comment:
Right now the logic for the package restore is as follows:
1. We update the Pharmaverse channel repo before restoring the R packages so the renv knows where to download the teal packages from.
2. Just getting into an R shell in the app directory should start the `renv::restore()`
3. Note that the `renv.lock` file does not contain any teal packages because we do not want to maintain fixed versions of these packages. renv automatically installs the app dependencies even if it is not present in the lock file, this will also make sure that the un-recorded packages are up to date.
4. After the `renv::restore()` that is done automatically we need to update the packages (CRAN and BioC, not teal packages) from the lock file and update the snapshot so that it keeps on updating.

Reply:
> Note that the renv.lock file does not contain any teal packages because we do not want to maintain fixed versions of these packages. renv automatically installs the app dependencies even if it is not present in the lock file, this will also make sure that the un-recorded packages are up to date.

(Haven't checked this) Does it mean that a simple `restore()` won't work? How about `restore_and_run`?

Reply:
Yes, That is the one drawback of this approach. But on running the `renv::status()` it will show the users that teal is not installed, not recorded, but used in the project. But this is a temporary solution as we move to CRAN the stable repo will contain snapshots of older versions so we can start to maintain them in the renv lock file.